doc/sphinx/chpl2rst.py

```diff
@@ -0,0 +1,189 @@
+                        rstline = rstline.strip()
+
+                # Strip indentation
+                if indentation == -1:
+                    lineindentation = len(rstline) - len(rstline.lstrip(' '))
+                    if lineindentation > 0:
+                        indentation = lineindentation
+                        rstline = rstline.lstrip(' ')
+                else:
+                    if rstline.startswith(' '*indentation):
+                        rstline = rstline[indentation:]
+                    else:
+                        rstline = rstline.lstrip(' ')
+
+                output.append(rstline)
+            else:
+                # Reset indentation
+                indentation = -1
+
+                # Write code block
+                if state == 'codeblock':
+                    output.append('')
+                    output.append('.. code-block:: chapel')
+                    output.append('')
+                codeline = ''.join(['    ', line])
+                output.append(codeline)
+
+    return '\n'.join(output)
+
+
+def chpl2codeblock(chapelfile):
+    """Convert Chapel file to giant code block rst file"""
+
+    filename = os.path.split(chapelfile)[1]
+    basename, _ = os.path.splitext(filename)
+    # Title
+    output = []
+    output.append(basename)
+    output.append('='*len(basename))
+    output.append('')
+    output.append('.. code-block:: chapel')
+    output.append('')
+
+    with open(chapelfile, 'r') as handle:
+        for line in [l.strip('\n') for l in handle.readlines()]:
+            output.append('  ' + line)
+
+    return '\n'.join(output)
+
+
+def getfname(chapelfile, output, prefix):
+    """Compute filename for output"""
+    if output == 'rst':
+        filename = os.path.split(chapelfile)[1]
+        basename, _ = os.path.splitext(filename)
+        rstname = ''.join([basename, '.rst'])
+        rstfile = os.path.join(prefix, rstname)
+        return rstfile
+    elif output == 'stdout':
+        print('Warning: prefix ignored for stdout')
+        return None
+    else:
+        print('Error: output = {0} is invalid')
+        sys.exit(1)
+
+
+def write(rstoutput, output):
+    """Write to output"""
+    if output is None:
+        sys.stdout.write(rstoutput)
+        return
+
+    with open(output, 'w') as handle:
+        handle.write(rstoutput)
+
+
+def main(chapelfiles, output='rst', prefix='.', codeblock=False, verbose=False):
+    """Driver function - convert each file to rst and write to output"""
+    for chapelfile in chapelfiles:
+
+        if codeblock:
+            rstoutput = chpl2codeblock(chapelfile)
+        else:
+            rstoutput = chpl2rst(chapelfile)
+
+        fname = getfname(chapelfile, output, prefix)
+        if verbose:
+            print('writing output of {0} to {1}'.format(chapelfile, fname))
+        write(rstoutput, fname)
+
+if __name__ == '__main__':
+    ARGS = get_arguments()
```

Comment:
I know... Just trusted my linter, which refers to [PEP 8](https://www.python.org/dev/peps/pep-0008/#constants) about global constants.